src/vif.jl

```diff
@@ -0,0 +1,124 @@
+        throw(ArgumentError("VIF not meaningful for models with only one non-intercept term"))
+    # NB: The correlation matrix is positive definite and hence invertible
+    #     unless there is perfect rank deficiency, hence the warning.
+    # NB: The determinate technique for GVIF could also be applied
+    #     columnwise (instead of Term-wise) here, but it wouldn't really
+    #     be any more efficient because this is a Symmetric matrix and computing
+    #     all those determinants has its cost. The determinants also hint at
+    #     how you could show equivalency, if you remember that inversion is solving
+    #     a linear system and that Cramer's rule -- which uses determinants --
+    #     can also a linear system
+    # so we want diag(inv(mm)) but directly computing inverses is bad.
+    # that said, these are typically small-ish matrices and this is Simple.
+    return diag(inv(mm))
+end
+
+"""
+    gvif(m::RegressionModel; scale=false)
+
+Compute the generalized variance inflation factor (GVIF).
+
+If `scale=true`, then each GVIF is scaled by the degrees of freedom
+for (number of coefficients associated with) the predictor: ``GVIF^(1 / (2*df))``
+
+Returns a vector of inflation factors computed for each term except
+for the intercept.
+In other words, the corresponding coefficients are `termnames(m)[2:end]`.
+
+The [generalized variance inflation factor (VIF)](https://doi.org/10.2307/2290467)
+measures the increase in the variance of a (group of) parameter's estimate in a model
+with multiple parameters relative to the variance of a parameter's estimate in a
+model containing only that parameter. For continuous, numerical predictors, the GVIF
+is the same as the VIF, but for categorical predictors, the GVIF provides a single
+number for the entire group of contrast-coded coefficients associated with a categorical
+predictor.
+
+See also [`termnames`](@ref), [`vif`](@ref).
+
+!!! warning
+    This method will fail if there is (numerically) perfect multicollinearity,
+    i.e. rank deficiency (in the fixed effects). In that case though, the VIF
+    isn't particularly informative anyway.
+
+## References
+
+Fox, J., & Monette, G. (1992). Generalized Collinearity Diagnostics.
+Journal of the American Statistical Association, 87(417), 178. doi:10.2307/2290467
+"""
+function gvif(m::RegressionModel; scale=false)
+    mm = StatsBase.cov2cor!(vcov(m), stderror(m))
+
+    StatsModels.hasintercept(formula(m)) ||
+        throw(ArgumentError("GVIF only defined for models with an intercept term"))
+    mm = @view mm[2:end, 2:end]
+    size(mm, 2) > 1 ||
+        throw(ArgumentError("GVIF not meaningful for models with only one non-intercept term"))
+
+    tn = @view termnames(m)[2:end]
+    trms = @view _terms(m)[2:end]
+
+    df = width.(trms)
+    vals = zeros(eltype(mm), length(tn))
+    # benchmarking shows thad adding in Symmetric() here before det()
+    # actually slows things down a bit
+    detmm = det(mm)
```

Comment:
Just using `Symmetric` will make `det` use the Bunch-Kaufman factorization for indefinite matrices instead of the LU and typically the LU will be more optimized. What *should* be faster is `det(cholesky(Symmetric(mm)))`.